src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -0,0 +1,267 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import logging
+from collections import defaultdict
+from typing import List, Optional
+
+from sortedcontainers import SortedList
+
+from snowflake.snowpark._internal.analyzer.analyzer_utils import (
+    drop_table_if_exists_statement,
+)
+from snowflake.snowpark._internal.analyzer.binary_plan_node import Union
+from snowflake.snowpark._internal.analyzer.query_plan_analysis_utils import (
+    get_complexity_score,
+)
+from snowflake.snowpark._internal.analyzer.select_statement import (
+    SET_UNION,
+    Selectable,
+    SelectSnowflakePlan,
+    SelectStatement,
+    SelectTableFunction,
+    SetStatement,
+)
+from snowflake.snowpark._internal.analyzer.snowflake_plan import Query, SnowflakePlan
+from snowflake.snowpark._internal.analyzer.snowflake_plan_node import (
+    LogicalPlan,
+    SaveMode,
+    SnowflakeCreateTable,
+    SnowflakeTable,
+    TableCreationSource,
+)
+from snowflake.snowpark._internal.analyzer.unary_plan_node import (
+    Aggregate,
+    CreateDynamicTableCommand,
+    CreateViewCommand,
+    Pivot,
+    Sample,
+    Sort,
+    Unpivot,
+)
+from snowflake.snowpark._internal.compiler.query_generator import QueryGenerator
+from snowflake.snowpark._internal.compiler.utils import (
+    TreeNode,
+    is_active_transaction,
+    replace_child,
+    update_resolvable_node,
+)
+from snowflake.snowpark._internal.utils import (
+    TempObjectType,
+    random_name_for_temp_object,
+)
+from snowflake.snowpark.session import Session
+
+COMPLEXITY_SCORE_LOWER_BOUND = 10_000_000
```

Comment:
add a comment here about the reference to the server side parameter value about the node count